main/lsp/lsp.h

```diff
@@ -238,26 +251,34 @@ class LSPLoop {
+                                       UnorderedMap<std::string, SorbetWorkspaceFileUpdate> &updates) const;
     void preprocessSorbetWorkspaceEdit(const DidCloseTextDocumentParams &closeParams,
-                                       UnorderedMap<std::string, std::string> &updates);
+                                       UnorderedMap<std::string, SorbetWorkspaceFileUpdate> &updates) const;
     void preprocessSorbetWorkspaceEdit(const WatchmanQueryResponse &queryResponse,
-                                       UnorderedMap<std::string, std::string> &updates);
-    LSPResult handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
-                                        const DidChangeTextDocumentParams &changeParams);
-    LSPResult handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
-                                        const DidOpenTextDocumentParams &openParams);
-    LSPResult handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
-                                        const DidCloseTextDocumentParams &closeParams);
-    LSPResult handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
-                                        const WatchmanQueryResponse &queryResponse);
-    LSPResult handleSorbetWorkspaceEdits(std::unique_ptr<core::GlobalState> gs,
-                                         std::vector<std::unique_ptr<SorbetWorkspaceEdit>> &edits);
-    LSPResult commitSorbetWorkspaceEdits(std::unique_ptr<core::GlobalState> gs,
-                                         UnorderedMap<std::string, std::string> &updates);
+                                       UnorderedMap<std::string, SorbetWorkspaceFileUpdate> &updates) const;
+    TypecheckRun handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
+                                           const DidChangeTextDocumentParams &changeParams) const;
+    TypecheckRun handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
+                                           const DidOpenTextDocumentParams &openParams) const;
+    TypecheckRun handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
+                                           const DidCloseTextDocumentParams &closeParams) const;
+    TypecheckRun handleSorbetWorkspaceEdit(std::unique_ptr<core::GlobalState> gs,
+                                           const WatchmanQueryResponse &queryResponse) const;
+    TypecheckRun handleSorbetWorkspaceEdits(std::unique_ptr<core::GlobalState> gs,
+                                            std::vector<std::unique_ptr<SorbetWorkspaceEdit>> &edits) const;
+    TypecheckRun commitSorbetWorkspaceEdits(std::unique_ptr<core::GlobalState> gs,
+                                            UnorderedMap<std::string, SorbetWorkspaceFileUpdate> &updates) const;
+    static std::string_view getFileContents(UnorderedMap<std::string, LSPLoop::SorbetWorkspaceFileUpdate> &updates,
```

Comment:
it seems that a bunch of helper methods below `hasSimilarName` & others can also be made `const`.

Reply:
? this methid is `static`. `hasSimilarName` and friends aren't class methods, they are just in the namespace -- perhaps they should be static methods on the class instead.

Reply:
/me got confused. you're right